### PR TITLE
MAT: remaining card scripts

### PR DESCRIPTION
**Pull request by Northmoc**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/undercity_upheaval.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Name:Undercity Upheaval
+ManaCost:1 G G
+Types:Sorcery
+A:SP$ PutCounter | ValidTgts$ Creature.YouCtrl | TgtPrompt$ Select any number of creatures you control to distribute counters to | CounterType$ P1P1 | CounterNum$ X | DividedAsYouChoose$ X | TargetMin$ 1 | TargetMax$ X | SubAbility$ DBPump | SpellDescription$ Distribute X +1/+1 counters among any number of target creatures you control, where X is the number of creature cards in your graveyard as you cast this spell.
+SVar:DBPump:DB$ PumpAll | ValidCards$ Creature.YouCtrl | KW$ Vigilance | SpellDescription$ Creatures you control gain vigilance until end of turn.
+SVar:X:Count$LastStateGraveyardWithFallback Creature.YouOwn
+DeckHints:Ability$Discard|Sacrifice
+Oracle:Undergrowth — Distribute X +1/+1 counters among any number of target creatures you control, where X is the number of creature cards in your graveyard as you cast this spell. Creatures you control gain vigilance until end of turn.
```
Comment (Agetian) [inline forge-gui/res/cardsfolder/upcoming/undercity_upheaval.txt:8]: DeckHas:Ability$Counters

File: forge-gui/res/cardsfolder/upcoming/tolarian_contempt.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Name:Tolarian Contempt
+ManaCost:3 U U
+Types:Enchantment
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigPutCounterAll | TriggerDescription$ When CARDNAME enters the battlefield, put a rejection counter on each creature your opponents control.
+SVar:TrigPutCounterAll:DB$ PutCounterAll | ValidCards$ Creature.OppCtrl | CounterType$ REJECTION
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigTarget | TriggerDescription$ At the beginning of your end step, for each opponent, choose up to one target creature they control with a rejection counter on it. That creature's owner puts it on the top or bottom of their library.
+SVar:TrigTarget:DB$ Pump | ValidTgts$ Creature.counters_GE1_REJECTION | TgtPrompt$ Select up to one target creature each opponent controls with a rejection counter on it | TargetMin$ 0 | TargetMax$ PlayerCountOpponents$Amount | TargetsWithDifferentControllers$ True | SubAbility$ DBRepeat | IsCurse$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/tolarian_contempt.txt:7]: `+OppCtrl`

File: forge-gui/res/cardsfolder/upcoming/undercity_upheaval.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Name:Undercity Upheaval
+ManaCost:1 G G
+Types:Sorcery
+A:SP$ PutCounter | ValidTgts$ Creature.YouCtrl | TgtPrompt$ Select any number of creatures you control to distribute counters to | CounterType$ P1P1 | CounterNum$ X | DividedAsYouChoose$ X | TargetMin$ 1 | TargetMax$ X | SubAbility$ DBPump | SpellDescription$ Distribute X +1/+1 counters among any number of target creatures you control, where X is the number of creature cards in your graveyard as you cast this spell.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/undercity_upheaval.txt:4]: @Hanmac 
We need to update such cards with `TargetMin$ 0` ?

> You may choose zero target creatures for Yannik’s reflexive triggered ability. If you do, no creature receives +1/+1 counters.

File: forge-gui/res/cardsfolder/upcoming/vesuvan_drifter.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Name:Vesuvan Drifter
+ManaCost:2 U
+Types:Creature Shapeshifter
+PT:2/4
+K:Flying
+S:Mode$ Continuous | Affected$ Card.TopLibrary+YouCtrl | AffectedZone$ Library | MayLookAt$ You | Description$ You may look at the top card of your library any time.
+T:Mode$ Phase | Phase$ BeginCombat | TriggerZones$ Battlefield | Execute$ TrigReveal | OptionalDecider$ You | TriggerDescription$ At the beginning of each combat, you may reveal the top card of your library. If you reveal a creature card this way, CARDNAME becomes a copy of that card until end of turn, except it has flying.
+SVar:TrigReveal:DB$ PeekAndReveal | NoPeek$ True | RememberRevealed$ True | SubAbility$ DBClone
+SVar:DBClone:DB$ Clone | ConditionDefined$ Remembered | ConditionPresent$ Creature | Defined$ Remembered | CloneTarget$ Self | Duration$ UntilEndOfTurn | AddKeywords$ Flying
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/vesuvan_drifter.txt:9]: Can just use `TopOfLibrary` instead of remembering...?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/vesuvan_drifter.txt]: The condition should still be checked

Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/vesuvan_drifter.txt]: 🤦‍♂️

**Review by tool4ever — APPROVED** [review]

**Pull request merged**